src/toplevel_generators.jl

```diff
@@ -197,6 +203,7 @@ macro generate_main_file_header(src_name)
         generate_compiler_options(src_name),
         # import Artifacts module
         generate_imports(src_name),
+        global_typeassert_available ? :(global artifact_dir::String) : :()
```

Comment:
This is here just so the `global artifact_dir` call in the `__init__` function will have a known type.